src/ngamsPlugIns/ngamsPlugIns/ngamsCmd_HTTPFETCH.py

```diff
@@ -98,28 +100,34 @@ def save_to_file(ngams_server, request_properties, target_filename, block_size,
     logger.info('Fetching file ID %s, checksum %s, checksum variant %s', file_id, checksum, crc_variant)
     crc_info = ngamsFileUtils.get_checksum_info(crc_variant)
     if start_byte != 0:
-        logger.info("resume requested")
+        logger.info("Resume requested from start byte %d", start_byte)
+
     if start_byte != 0 and download_resume_supported:
-        logger.info("Resume requested and mirroring source supports resume. Appending data to previously started staging file")
+        logger.info("Resume requested and mirroring source supports resume. Appending data to previous staging file")
         crc = ngamsFileUtils.get_checksum(65536, target_filename, crc_variant)
         request_properties.setBytesReceived(start_byte)
-        fd_out = open(target_filename, "a")
+        fd_out = open(target_filename, "ab")
     else:
         if start_byte > 0:
             logger.info("Resume of download requested but server does not support it. Starting from byte 0 again.")
-        fd_out = open(target_filename, "w")
+        fd_out = open(target_filename, "wb")
         crc = crc_info.init
 
     fetch_start_time = time.time()
 
-    # Distinguish between Archive Pull and Push Request. By Archive pull we may simply read the file descriptor until
-    # it returns "".
-    logger.info("It is an HTTP Archive Pull Request: trying to get Content-length")
-    hdrs = {h[0]: h[1] for h in response.getheaders()}
-    if 'content-length' in hdrs:
-        remaining_size = int(hdrs['content-length'])
+    # Distinguish between archive pull and push request
+    # By archive pull we may simply read the file descriptor until it returns and empty string
+    response_header_dict = {h[0]: h[1] for h in response.getheaders()}
+    if 'content-length' in response_header_dict:
+        # For some reason python 2 uses lower case 'content-length'
+        remaining_size = int(response_header_dict['content-length'])
+        logger.debug("Got Content-Length header value %d in response", remaining_size)
+    elif 'Content-Length' in response_header_dict:
+        # For some reason python 3 uses mixed case 'Content-Length'
+        remaining_size = int(response_header_dict['Content-Length'])
+        logger.debug("Got Content-Length header value %d in response", remaining_size)
```

Comment:
What about doing `response_header_dict = {h[0].lower(): h[1] ...}`, then you only need the first `if`.

Reply:
I have pushed a commit following your suggestion